File: python/allergies/allergies.py
class Allergies(object):
    known = [
        'eggs', 'peanuts', 'shellfish', 'strawberries',
        'tomatoes', 'chocolate', 'pollen', 'cats'
    ]

    def __init__(self, score):
        self.lst = [n for i, n in enumerate(self.known) if score & (1 << i)]

    def is_allergic_to(self, item):
        return item in self.lst

File: zig/allergies/allergies.zig
const std = @import("std");
const EnumSet = std.EnumSet;

pub const Allergen = enum {
    eggs,
    peanuts,
    shellfish,
    strawberries,
    tomatoes,
    chocolate,
    pollen,
    cats,
};

pub fn isAllergicTo(score: u8, allergen: Allergen) bool {
    const allergies = initAllergenSet(score);
    return allergies.contains(allergen);
}

pub fn initAllergenSet(score: usize) EnumSet(Allergen) {
    return .{ .bits = .{ .mask = @truncate(score) } };
}
